### ISSUE TITLE
Shift+Enter does not work reliably in the editor in a multi-threaded application

### ISSUE THREAD

**Issue by mkromberg** [Dyalog/ride#627]
### Describe the issue you are having
When editing a function in a multi-threaded application, Shift+Enter does not work reliably.

### Did you connect to an already running interpreter or start the interpreter from RIDE?
Does not matter.

### How do you reproduce the issue?
Follow the instructions from issue 626, but instead of tracing foo, edit it using )ed foo. Put the cursor over the name Init. Shift+Enter will sometimes open the function Init, and sometimes not.

### Paste the contents of Help → About (Shift+F1)
```
IDE:
  Version: 4.4.3469
  Platform: Win32
  Date: 2020-07-07 16:02:26 +0100
  Git commit: b97d35464cceaf671e05e82e8db59e50f69a0481
  Preferences:{
    "connectFav":"1",
    "floatSingle":"0",
    "floating":"1",
    "kbdLocale":"da_DK",
    "prefixKey":"<"
  }

Interpreter:
  Version: 19.0.39694
  Platform: Linux-64
  Edition: Unicode/64
  Date: Dec  8 2020 at 03:38:37
```

**Comment by mkromberg** [comment]
The situation is now much improved, but there is still an issue which is presumed to be interpreter-side (Mantis 18833).

**Comment by e9gille** [comment]
Fixed by #649
